src/service-implementations/lf-repo-tree-node-service.ts

```diff
@@ -272,55 +281,104 @@ export class LfRepoTreeNodeService implements LfTreeNodeService {
    * };
    *  ```
    */
-  createLfRepoTreeNode(entry: Entry, repoName: string, parent?: LfRepoTreeNode): LfRepoTreeNode {
+  createLfRepoTreeNodeWithParent(entry: Entry, parent: LfRepoTreeNode): LfRepoTreeNode {
     let treeNode: LfRepoTreeNode | undefined;
     if (!entry.entryType) {
       throw new Error('entry type is undefined');
     }
-    if (entry.entryType === EntryType.Folder) {
-      treeNode = this.createFolderNode(entry, repoName, parent);
-    } else if (entry.entryType === EntryType.Document) {
-      treeNode = this.createLeafNode(entry, parent);
-    } else if (entry.entryType === EntryType.Shortcut) {
-      const shortcut = entry as Shortcut;
-      if (shortcut.targetType === EntryType.Folder) {
-        treeNode = this.createFolderNode(shortcut, repoName, parent);
-      } else if (shortcut.targetType === EntryType.Document) {
-        treeNode = this.createLeafNode(shortcut, parent);
+    switch (entry.entryType) {
+      case EntryType.Folder:
+        treeNode = this.createFolderNodeWithParent(entry, parent);
+        break
+      case EntryType.RecordSeries:
+        treeNode = this.createFolderNodeWithParent(entry, parent);
+        break
+      case EntryType.Document:
+        treeNode = this.createLeafNode(entry, parent);
+        break
+      case EntryType.Shortcut:
+          switch ((entry as Shortcut).targetType) {
+            case EntryType.Folder:
+              treeNode = this.createFolderNodeWithParent(entry as Shortcut, parent);
+              break
```

Comment:
if this is the same as the above switch, refactor the switch 
maybe instead have an if (shortcut) first which determines whether to use entry or targetType and then pass that in to the function with the switch based on Folder, Document, REcordSEries, etc.

you shouldn't have to case entry as Shortcut when passing as a parameter, all data will get passed through